### ISSUE TITLE
Allow adding fields with checkbox in Additional Fields API.

### ISSUE THREAD

**Issue by senadir** [woothemes/woocommerce#42080]
A portion of custom fields introduced are usually conditional fields, meaning you need to check a checkbox before you can see and fill the field. For example:

- Add a gift note? which would render a gift input.
- Add an order note? which renders the order note text area.
- Shopping as a business? which would render a VAT field.

This would remove the need to register two fields or create a custom bulk, and would bulk them into a single one.

For now, this would only be limited to textual fields and select fields, you can't have a checkbox that would show yet another checkbox.

**Comment by senadir** [comment]
@pmcpinto @ralucaStan what do you think about this? do you feel like we need it? it can exists on a future iteration (not a must go right now) but just checking if it make sense or will it pollute the UI.

**Comment by elizaan36** [comment]
@senadir Could you clarify the design needs on this one?

**Comment by senadir** [comment]
Hey @elizaan36 mainly prototyping how would this look like in Checkout. We have 3 uses cases with a couple of examples:

- Order note is behind a checkbox.
- Add a coupon and add a gift card are both behind toggle as well.

**Comment by elizaan36** [comment]
Aren't these features offered by core? I'd be concerned about duplication when a 3PD adds a field or element that already exists.

**Comment by senadir** [comment]
Yes they're. We don't expect people to recreate core fields, but they're example of a how a text field can have a checkbox opt in.

**Comment by webdados** [comment]
How is this going? Any developments?

I now have at least two websites where we need to implement this or #47833.
As a mater of fact, I think #47833 could solve both issues. (Hide another field based on the checkbox status)

**Comment by pierorocca** [comment]
Closing as completed.
